src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -896,6 +907,49 @@ private void EnsureRavenEntityName(BlittableJsonReaderObject obj,string collecti
             }
             return true;
         }
+        
+        private bool TryResolveByLeader(
+            DocumentsOperationContext context,
+            string key)
+        {
+            var leader = _parent.ResolverLeader;
+            if (leader?.Dbid == default(Guid))
+            {
+                return false;
+            }
+
+            var conflicts = _database.DocumentsStorage.GetConflictsFor(context, key);
+            DocumentConflict resolved = null;
+            long maxEtag = -1;
+            foreach (var documentConflict in conflicts)
+            {
+                foreach (var changeVectorEntry in documentConflict.ChangeVector)
+                {
+                    if (changeVectorEntry.DbId.Equals(leader.Dbid))
+                    {
+                        if (changeVectorEntry.Etag == maxEtag)
+                        { 
+                            // we have two documents with same etag of the leader
+                            return false;
+                        }
+
+                        if (changeVectorEntry.Etag < maxEtag)
+                            continue;
+
+                        maxEtag = changeVectorEntry.Etag;
+                        resolved = documentConflict;
+                        break;
+                    }
+                }
+            }
+
+            if (resolved == null)
+                return false;
+    
+            _database.DocumentsStorage.Put(context, key, null, resolved.Doc.Clone(context));
```

Comment:
The resolved.Doc points to a conflicted document, which we delete inside the Put method. Before we actually putting the document.